pkg/prometheus/promcfg.go

```diff
@@ -2142,6 +2151,44 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 	}
 }
 
+func generateRoleSelectorConfig(role string, selector *metav1.LabelSelector, k8sSDConfig yaml.MapSlice, cg *ConfigGenerator) yaml.MapSlice {
```

Comment:
can we use https://pkg.go.dev/k8s.io/apimachinery@v0.31.3/pkg/labels to generate the selector string instead of building it ourselves?